data/src/main/java/com/linkedin/data/schema/grammar/PdlSchemaParser.java

```diff
@@ -489,9 +583,24 @@ else if (namedType.enumDeclaration() != null)
     }
 
     schema.setPackage(getCurrentPackage());
+    recordLocation(schema, namedType);
     return schema;
   }
 
+  private void recordLocation(Object schemaNode, ParserRuleContext context)
+  {
+    if (_isLocationNeeded)
+    {
+      // getCharPosition returns beginning of the last token. Add the token's length to get actual end position.
+      int endPosition = context.getStop().getCharPositionInLine() +
+          (context.getStop().getStopIndex() - context.getStop().getStartIndex());
+      // Parser columns are indexed at 0, so add 1 to get it 1 indexed.
+      _parseLocations.put(schemaNode,
```

Comment:
So what is a schemaNode? 
Is it the whole PDL file record or the smallest unit on the schema (like the field)
If it is the whole file then there seems to be only one location allowed to be recorded?

Reply:
schema node (I'll rename it as element) is any "referencable" element in the schema. It includes fields, union members, inline schemas, enum symbols etc. For the full list, look at the java doc of getParseLocations method.